--- a/src/wifi.vala
+++ b/src/wifi.vala
@@ -49,6 +49,7 @@

         foreach (var ap in device.access_points) {
             var new_ap = new AccessPoint(this, ap);
+            if (ap.bssid == null) continue;
             _access_points.set(ap.bssid, new_ap);
             access_point_added(new_ap);
         }
@@ -56,6 +57,7 @@
         device.access_point_added.connect((access_point) => {
             var ap = (NM.AccessPoint)access_point;
             var new_ap = new AccessPoint(this, ap);
+            if (ap.bssid == null) return;
             _access_points.set(ap.bssid, new_ap);
             access_point_added(new_ap);
             notify_property("access-points");
@@ -63,6 +65,7 @@

         device.access_point_removed.connect((access_point) => {
             var ap = (NM.AccessPoint)access_point;
+            if (ap.bssid == null) return;
             var rem_ap = _access_points.get(ap.bssid);
             _access_points.remove(ap.bssid);
             access_point_removed(rem_ap);
@@ -151,7 +154,7 @@

         var ap = device.active_access_point;
-        if (ap != null) {
+        if (ap != null && ap.bssid != null) {
             active_access_point = _access_points.get(ap.bssid);
             on_active_access_point_notify();
             ap_handler = active_access_point.notify.connect(on_active_access_point_notify);
